src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -320,14 +323,6 @@ int xocl_program_shell(struct xocl_dev *xdev, bool force)
 		goto failed;
 	}
 
-	if (xdev->core.fdt_blob) {
-		vfree(xdev->core.fdt_blob);
-		xdev->core.fdt_blob = NULL;
```

Comment:
where do you do the clean up now?

Reply:
1. resolve deadlock to offline mailbox in mailbox handler
2. It is more safe to sync with hot reset request.
We mgmtpf completes PLP programming, it notify userpf. Userpf will call refresh which free the current blob and use the new one.

Reply:
ok